code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -236,47 +297,57 @@
 	D.visible_message("<span class='warning'>\The [D]'s motors whine as something attempts to brute force their way through it!</span>")
 	playsound(get_turf(D), 'sound/effects/grillehit.ogg', 50, 1)
 	D.shake(1, 8)
-	if(do_after(src, D, 100*time_mult, needhand = FALSE)) //Roughly 10 seconds for people on the other side of the door to run the heck away
-		if(can_open_door(D))//Let's see if nobody quickly bolted it
-			if(break_doors == CANPLUS) //Guaranteed
-				D.visible_message("<span class='warning'>\The [D] breaks open under the pressure</span>")
-				if(istype(D, /obj/machinery/door/airlock/))
-					var/obj/machinery/door/airlock/A = D
-					A.locked = 0
-					A.welded = 0
-					A.jammed = 0
-				D.open(1)
-			else
-				if(prob(33))
-					D.visible_message("<span class='warning'>\The [D] creaks open under force, steadily</span>")
+	busy = TRUE
+	var/target_loc = D.loc
+	var/self_loc = src.loc
+	spawn(100*time_mult)
```

Comment:
Yes, seconds is just defined as `* 10`. It's a sneaky Byond trick for legibility.